### DISCUSSION TITLE
How do I open a workspace on launch?

### DISCUSSION THREAD

**Discussion by stevefan1999-personal** [gitpod-io/openvscode-server#247] [Q&A] [answered]
So I wanted to set a default workspace path to open on root path (`/`). 

Previously, we can launch vscode by `code /home` and such, but since openvscode-server uses href, this means it have to append the workspace url parameter which is [here](https://github1s.com/gitpod-io/openvscode-server/blob/HEAD/src/vs/code/browser/workbench/workbench.ts#L299-L300) which is not very ergonomic since we sit behind a reverse proxy.

Which settings influences the default workspace path other than setting on href as well?

**Comment by jeanp413** [comment, marked as answer]
@stevefan1999-personal  There are `--folder` and `--workspace` arguments that you can pass when you launch the server `server.sh`

**Reply by santiagobasulto** [reply]
Thanks @jeanp413. This was also helpful for me. Is it documented anywhere?

**Reply by jeanp413** [reply]
I think you can pass `--help` to the server script to get a summary of the available options.
In the code you can find them here https://github.com/gitpod-io/openvscode-server/blob/main/src/vs/server/serverEnvironmentService.ts

**Reply by elzapp** [reply]
`--help` is not a supported flag :-(

**Comment by sevetseh28** [comment]
I see this is not supported by the binary `openvscode-server`. What script is that @jeanp413 ?

**Comment by l0rd** [comment]
Options `--folder` and `--workspace` [aren't supported anymore on the server side](https://github.com/gitpod-io/openvscode-server/blob/d7e8cee635409ab0ee3c87331d981ee3e4f1f067/src/vs/server/node/serverEnvironmentService.ts#L46). `--default-folder` and `--default-workspace` can be used instead.
